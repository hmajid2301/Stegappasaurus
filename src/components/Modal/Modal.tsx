import React from 'react';
import {Modal, TouchableOpacity, View} from 'react-native';
import {Icon} from 'react-native-elements';
import styled from 'styled-components/native';

import {bodyLight} from '~/constants/fonts';
import {ThemeColors} from '~/constants/types';
import {ThemeContext} from '~/providers/ThemeContext';

interface Props {
  children: JSX.Element;
  name: string;
}

interface State {
  isVisible: boolean;
}

export default class MyModal extends React.Component<Props, State> {
  public static contextType = ThemeContext;
  public context!: React.ContextType<typeof ThemeContext>;

  public state: State = {
    isVisible: false,
  };

  public render() {
    const {background, color} = this.context.theme;
    return (
      <View>
        <Modal
          animationType="slide"
          onRequestClose={this.setModalVisibility.bind(this, false)}
          transparent={false}
          visible={this.state.isVisible}>
          <ModalContainer background={background}>
            <TouchableIcon
              onPress={this.setModalVisibility.bind(
                this,
                !this.state.isVisible,
              )}
              testID="close">
              <Icon iconStyle={{color}} name="close" type="evil-icons" />
            </TouchableIcon>
            {this.props.children}
          </ModalContainer>
        </Modal>

        <TouchableOpacity onPress={this.setModalVisibility.bind(this, true)}>
          <TouchableText color={color}>{this.props.name}</TouchableText>
        </TouchableOpacity>
      </View>
    );
  }

  private setModalVisibility(visible: boolean) {
    this.setState({isVisible: visible});
  }
}

const TouchableText = styled.Text<{color: ThemeColors}>`
  color: ${props => props.color};
  font-family: ${bodyLight};
`;

const ModalContainer = styled.View<{background: ThemeColors}>`
  flex: 1;
  background-color: ${props => props.background};
`;

const TouchableIcon = styled.TouchableOpacity`
  align-self: flex-end;
  padding: 10px;
`;
