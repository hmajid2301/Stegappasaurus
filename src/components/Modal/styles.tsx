import {StyleSheet} from 'react-native';

import {bodyLight} from '~/constants/fonts';

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },

  iconContainer: {
    alignSelf: 'flex-end',
    paddingHorizontal: 10,
    paddingVertical: 10,
  },

  buttonText: {
    fontFamily: bodyLight,
  },
});

export default styles;
