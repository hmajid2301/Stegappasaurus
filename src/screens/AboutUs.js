import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { View, Text } from 'react-native';
import { Icon } from 'react-native-elements';

import CustomHeader from '../components/CustomHeader';
import colors from '../components/Common/colors';


export default class AboutUs extends Component {
  static navigationOptions = {
    drawerLabel: 'About Us',
    drawerIcon: ({ tintColor }) => (
      <Icon name='info' type='simple-line-icons' color={tintColor} />
    ),
  };

  static propTypes = {
    navigation: PropTypes.shape({
      navigate: PropTypes.func.isRequired,
    }).isRequired,
  };

  render() {
    return (
      <View>
        <CustomHeader navigation={this.props.navigation} color={colors.quaternaryColor}/>
        <Text> About Us </Text>
      </View>
    );
  }
}
