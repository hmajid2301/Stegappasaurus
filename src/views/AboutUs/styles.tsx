import { StyleSheet } from "react-native";

import { fonts } from "~/modules";

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "column"
  },

  textContainer: {
    alignItems: "center",
    paddingVertical: 20
  },

  about: {
    fontFamily: fonts.bodyLight,
    paddingHorizontal: 20
  }
});

export default styles;
