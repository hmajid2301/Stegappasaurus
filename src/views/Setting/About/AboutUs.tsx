import React, {useContext} from 'react';
import {ScrollView, View} from 'react-native';
import styled from 'styled-components/native';

import AboutList from '~/components/AboutList/AboutList';
import {bodyLight} from '~/constants/fonts';
import {ThemeColors} from '~/constants/types';
import {about} from '~/data';
import {ThemeContext} from '~/providers/ThemeContext';

const AboutUs = () => {
  const {background, color} = useContext(ThemeContext).theme;

  return (
    <AboutContainer background={background}>
      <ScrollView>
        <TextContainer>
          <AboutText color={color}>
            Stegappasaurus is a free mobile application fully open source, built
            using React Native. This application uses steganography algorithms
            to hide your (text) data within images. This project was originally
            created as third year project for University. However this version
            is a complete rewrite of the application.
          </AboutText>
        </TextContainer>

        <View>
          <AboutList items={about(color)} />
        </View>
      </ScrollView>
    </AboutContainer>
  );
};

const AboutContainer = styled.View<{background: ThemeColors}>`
  background: ${props => props.background};
  flex: 1;
  flex-direction: column;
`;

const TextContainer = styled.View`
  align-items: center;
  padding: 20px 0;
`;

const AboutText = styled.Text<{color: ThemeColors}>`
  color: ${props => props.color};
  font-family: ${bodyLight};
  padding: 0 20px;
`;

export default AboutUs;
