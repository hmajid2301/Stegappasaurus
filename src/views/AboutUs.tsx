import React from "react";
import { ScrollView, Text, View } from "react-native";
import { NavigationScreenProp } from "react-navigation";

import { ITheme } from "@types";
import AboutList from "~/components/AboutList";
import AppHeader from "~/components/AppHeader";
import { about } from "~/data";
import styles from "~/views/AboutUs/styles";

interface IProps {
  navigation: NavigationScreenProp<any, any>;
  screenProps: {
    theme: ITheme;
  };
}

export default class AboutUs extends React.Component<IProps, {}> {
  public render() {
    const { theme } = this.props.screenProps;

    return (
      <View style={[styles.container, { backgroundColor: theme.background }]}>
        <ScrollView>
          <AppHeader navigation={this.props.navigation} theme={theme} />

          <View style={styles.textContainer}>
            <Text style={[styles.about, { color: theme.color }]}>
              This project was developed by me as a way to learn React Native.
              It is a complete rewrite of the my university project, which was
              written in Ionic (Apache Cordova). This mobile application allow
              you, the user, to encode image with text using steganography
              algorithms.
            </Text>
          </View>

          <View>
            <AboutList
              items={about}
              backgroundColor={theme.background}
              color={theme.color}
            />
          </View>
        </ScrollView>
      </View>
    );
  }
}
