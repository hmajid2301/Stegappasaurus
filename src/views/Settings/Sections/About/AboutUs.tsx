import React from 'react';
import {ScrollView, Text, View} from 'react-native';

import AboutList from '~/components/AboutList';
import {ThemeColors} from '~/constants/types';
import {about} from '~/data';
import styles from './AboutUs/styles';

interface IProps {
  background: ThemeColors;
  color: ThemeColors;
}

const AboutUs = (props: IProps) => (
  <View style={[styles.container, {backgroundColor: props.background}]}>
    <ScrollView>
      <View style={styles.textContainer}>
        <Text style={[styles.about, {color: props.color}]}>
          Stegappasaurus is a free mobile application fully open source, built
          using React Native. This application uses steganography algorithms to
          hide your (text) data within images. This project was originally
          created as third year project for University. However this version is
          a complete rewrite of the application.
        </Text>
      </View>

      <View>
        <AboutList
          items={about}
          backgroundColor={props.background}
          color={props.color}
        />
      </View>
    </ScrollView>
  </View>
);

export default AboutUs;
