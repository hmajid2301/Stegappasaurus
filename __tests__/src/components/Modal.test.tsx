import {mount, shallow} from 'enzyme';
import React from 'react';

import Modal from '~/components/Modal';
import AboutUs from '~/views/Settings/Sections/About/AboutUs';

describe('odal: Match snapshots', () => {
  test('1', () => {
    const component = shallow(
      <Modal background="#17212D" color="#FFF" name="About Us">
        <AboutUs background="#17212D" color="#FFF" />
      </Modal>,
    );
    expect(component).toMatchSnapshot();
  });
});

describe('Modal Props', () => {
  test('Modal: onRequestClose', () => {
    const component = mount(
      <Modal background="#17212D" color="#FFF" name="FAQ">
        <AboutUs background="#17212D" color="#FFF" />
      </Modal>,
    );

    (component
      .find('Modal')
      .first()
      .props() as any).onRequestClose();
    expect(component.state('isVisible')).toEqual(false);
  });

  test('TouchableOpacity: onPress', () => {
    const component = mount(
      <Modal background="#17212D" color="#FFF" name="FAQ">
        <AboutUs background="#17212D" color="#FFF" />
      </Modal>,
    );

    (component
      .find('TouchableOpacity')
      .first()
      .props() as any).onPress();
    expect(component.state('isVisible')).toEqual(true);
  });

  test('TouchableOpacity2: onPress', () => {
    const component = mount(
      <Modal background="#17212D" color="#FFF" name="FAQ">
        <AboutUs background="#17212D" color="#FFF" />
      </Modal>,
    );

    (component
      .find('TouchableOpacity')
      .last()
      .props() as any).onPress();
    expect(component.state('isVisible')).toEqual(true);
  });
});
