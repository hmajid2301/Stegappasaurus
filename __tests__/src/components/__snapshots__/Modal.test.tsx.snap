// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`odal: Match snapshots 1 1`] = `
<View>
  <Component
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={false}
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "#17212D",
          },
        ]
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignSelf": "flex-end",
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          }
        }
      >
        <Themed.Icon
          iconStyle={
            Object {
              "color": "#FFF",
            }
          }
          name="close"
          type="evil-icons"
        />
      </TouchableOpacity>
      <AboutUs
        background="#17212D"
        color="#FFF"
      />
    </View>
  </Component>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Roboto-Light",
          },
          Object {
            "color": "#FFF",
          },
        ]
      }
    >
      About Us
    </Text>
  </TouchableOpacity>
</View>
`;
